src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/NovelAdjacencyReferenceLocations.java

```diff
@@ -123,79 +138,150 @@ protected NovelAdjacencyReferenceLocations(final Kryo kryo, final Input input) {
-                    leftBreakpointRefContig  = ca.regionWithLowerCoordOnContig.referenceSpan.getContig();
-                    leftBreakpointCoord      = ca.regionWithLowerCoordOnContig.referenceSpan.getStart();
-                    rightBreakpointRefContig = ca.regionWithHigherCoordOnContig.referenceSpan.getContig();
-                    rightBreakpointCoord     = ca.regionWithHigherCoordOnContig.referenceSpan.getStart() + homologyLen;
-                } else {
-                    leftBreakpointRefContig  = ca.regionWithHigherCoordOnContig.referenceSpan.getContig();
-                    leftBreakpointCoord      = ca.regionWithHigherCoordOnContig.referenceSpan.getStart();
-                    rightBreakpointRefContig = ca.regionWithLowerCoordOnContig.referenceSpan.getContig();
-                    rightBreakpointCoord     = ca.regionWithLowerCoordOnContig.referenceSpan.getStart() + homologyLen;
+            }
+
+            private static boolean isFirstInPartner(final ChimericAlignment ca, final SAMSequenceDictionary referenceDictionary) {
+                switch (ca.strandSwitch) {
+                    case NO_SWITCH: return 0 > IntervalUtils.compareContigs(ca.regionWithLowerCoordOnContig.referenceSpan,
+                            ca.regionWithHigherCoordOnContig.referenceSpan, referenceDictionary);
+                    case FORWARD_TO_REVERSE: case REVERSE_TO_FORWARD:
+                        return ca.isForwardStrandRepresentation;
+                    default:
+                        throw new GATKException("Unseen strand switch case for: " + ca.onErrStringRep());
                 }
             }
         }
+    }
 
-        final SimpleInterval leftBreakpoint = new SimpleInterval(leftBreakpointRefContig, leftBreakpointCoord, leftBreakpointCoord);
-        final SimpleInterval rightBreakpoint = new SimpleInterval(rightBreakpointRefContig, rightBreakpointCoord, rightBreakpointCoord);
+    /**
+     * Returns the reference coordinates of the left and right breakpoints implied by this chimeric alignment.
+     * If there is homologous sequence represented in the alignments, it will be assigned to the side of the breakpoint
+     * with higher reference coordinates.
+     */
+    @VisibleForTesting
+    static Tuple2<SimpleInterval, SimpleInterval> leftJustifyBreakpoints(final ChimericAlignment ca,
+                                                                         final BreakpointComplications complication,
+                                                                         final SAMSequenceDictionary referenceDictionary) {
+
+        final boolean sameChromosome = ca.regionWithLowerCoordOnContig.referenceSpan.getContig()
+                .equals(ca.regionWithHigherCoordOnContig.referenceSpan.getContig());
+        final Tuple2<SimpleInterval, SimpleInterval> leftAdjustedBreakpoints;
+        if (sameChromosome) {
```

Comment:
This refactoring is almost there, but I think you could be helped out a lot by a factory method pattern:

make a static method `getContigToReferenceBreakpointMapping(ChimericAlignment, BreakpointComplications, SAMSequenceDictionary)` on `BreakpointLeftAligner` (or whatever you end up renaming that to, like `ContigToReferenceBreakpointMapping`) that returns a `BreakpointLeftAligner` (or `ContigToReferenceBreakpointMapping`). Then this method only calls `getLeftJustifiedBreakpoints` on the result. That way there's only one invocation of `getLeftJustifiedBreakpoints`, one method handles choosing the logic path to go down, and you have separate, self-contained places where the different chunks of logic are implemented.

Reply:
done. Learned factory pattern. Thanks!